pinot-query-planner/src/main/java/org/apache/pinot/query/catalog/PinotCatalog.java

```diff
@@ -75,7 +114,13 @@ public Table getTable(String name) {
    */
   @Override
   public Set<String> getTableNames() {
-    return _tableCache.getTableNameMap().keySet();
+    if (_databaseName != null) {
+      return _tableCache.getTableNameMap().keySet().stream()
+          .filter(n -> DatabaseUtils.isPartOfDatabase(n, _databaseName))
+          .collect(Collectors.toSet());
+    } else {
+      return Collections.emptySet();
```

Comment:
This means that the root schema cannot contain tables. Can we add a comment here?

Reply:
Will do